APIScrape/scrape.php

```diff
@@ -0,0 +1,176 @@
+                `img_src` varchar(255) NOT NULL DEFAULT '',
+                `breed_id` int(11) unsigned NOT NULL,
+                PRIMARY KEY (`id`),
+                UNIQUE KEY `img_src` (`img_src`)
+                ) ENGINE=InnoDB DEFAULT CHARSET=latin1;";
+        $db->exec($sql);
+        echo "Database successfully initialised.\n";
+    } catch (PDOException $e) {
+        echo $e->getMessage();
+    }
+}
+
+/**
+ * Make API request to get all of the breeds from theCatAPI.
+ * 
+ * @return array of breeds as 4-character breed id => breed name
+ * e.g. "ABYS" => "Abyssinian"
+ */
+function getCatBreeds():array
+{
+    $curl = curl_init();
+    curl_setopt_array($curl, array(
+        CURLOPT_URL => "https://api.thecatapi.com/v1/breeds/",
+        CURLOPT_CUSTOMREQUEST => "GET",
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_HTTPHEADER => [ API_KEY ],
+    ));
+    $response = curl_exec($curl);
+    if($response === false) {
+        echo 'Curl error: ' . curl_error($curl);
+    } else {
+        echo "Received list of cat breeds successfully.\n";
+    }
+    curl_close($curl);
+
+    $responseArray = json_decode($response, true);
+
+    $associativeBreedArray = [];
+    for ($i = 0; $i < count($responseArray); $i++) {
+        $associativeBreedArray[$responseArray[$i]["id"]] = $responseArray[$i]["name"];
+    }
+    return $associativeBreedArray;
+}
+
+/**
+ * Puts the list of breeds into the database.
+ * 
+ * @param PDO $db - the database to insert into.
+ * @param array $catBreeds - the list of cat breeds. As breed id => breed name.
+ * @return void
+ */
+function fillCatBreedToDB(PDO $db, array $catBreeds)
+{
+    foreach($catBreeds as $id => $breed) {
+        $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (:breedToString);');
+        $breedToString = "$breed";
+        $sql->bindParam('breedToString', $breedToString, PDO::PARAM_STR);
+        $sql->execute();
+    }
+    echo "Cat breeds added to database.\n";
+}
+
+/**
+ * Use the list of cat breeds to make the API requests so we can get URLs
+ * of cat pictures to link to in the app.
+ * 
+ * @param array of cat breeds
+ * @return array of breed names => array of strings representing the image URLs
+ */
+function getCatImageURLs(array $catBreeds):array
+{
+    $imageSourceArray = [];
+    $breedCount = count($catBreeds);
+    $i = 1;
+    echo "Fetching cat image URLs...\n";
+    foreach ($catBreeds as $id => $name) {
+        $catImageApiUrl = 'https://api.thecatapi.com/v1/images/search?breed_ids=' . $id . '&limit=21';
+        $curl = curl_init();
+        curl_setopt_array($curl, array(
+            CURLOPT_URL => $catImageApiUrl,
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_HTTPHEADER => [ API_KEY ]
+        ));
+        $imageApiResponse = curl_exec($curl);
+        if($imageApiResponse === false) {
+            echo 'Curl error: ' . curl_error($curl);
```

Comment:
This isn't handling the error, it's just showing me the error string and continuing the script without the data. Make the script exit on error.

Reply:
don't forget to do this below, too

Reply:
on it